### PR TITLE
feat: add modals

### LINKED ISSUE #228: request: add modals
**Is your feature request related to a problem? Please describe.**

Be up to date on Discord Interactions, such as **[modal interactions](https://discord.com/developers/docs/change-log#interaction-modals-and-application-command-attachment-option-type)**.

**Describe the solution you'd like**

Add support for modals and text input components. Basically, is a pop-up form and I'd like (if possible) to write on text inputs.

modal-example

It contains a Title, a warning, text inputs with a label, a 'Cancel' option and a 'Submit' button.

**Describe alternatives you've considered**
n/a

**Additional context**
n/a

### PR DESCRIPTION
**Pull request by eumarciel404**
Fixes #228 
Closes #419

### CODE REVIEW

**Comment by eumarciel404** [comment]
If the display  is default setting to flex, this stay in the front of all content
image

File: packages/react/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "@skyra/discord-components-react",
-	"version": "4.0.0-alpha.28",
+	"version": "4.0.0-alpha.29",
```
Comment (favna) [inline packages/react/package.json:3]:
```suggestion
	"version": "4.0.0-alpha.28",
```

File: packages/core/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
 	"name": "@skyra/discord-components-core",
-	"version": "4.0.0-alpha.28",
+	"version": "4.0.0-alpha.29",
```
Comment (favna) [inline packages/core/package.json:3]:
```suggestion
	"version": "4.0.0-alpha.28",
```

**Review by favna — CHANGES_REQUESTED** [review]
Versioning is handled by Lerna (and in the future by a GH action)

**Comment by eumarciel404** [comment]
> Versioning is handled by Lerna (and in the future by a GH action)

Okay, I didn't know

**Comment by favna** [comment]
I went through your modals PR and made some changes / fixes but there's still work to do as well. Namely the `discord-message-needed-input` (now called `discord-text-input-message-needed-input`) doesn't show properly. The problem is that its div is nested in the `DiscordInputText` when it should be nested in the `DiscordModal` so that it can actually overlap the `dialog`'s content.

As for what I changed:
- Refactored out all the `:any`'s
- You had some code comments in brazillian, I changed those to english
- There was a typo in `discord-input-text-container` CSS where you had `flex: collmun;` instead of `flex-direction: column;`
- Where you wrote `/* Here need danger color discord */` I got the proper colour
  - I also updated the colours to match Discord in a few other places
- I have `discord-message-needed-input` a `width 50%` because adding the border I noticed it was too short for the text
  - I updated the text to match Discord
- `_maxLengthCalc` and `_warn` were local values so I changed their decorator from `@property` to `@state` and made them `protected` because I want state to be accessible if someone decides to extend classes.
- I removed the nested ternaries for the `textarea` values, using `when` instead.
- I removed invalid `aria-labelledby` (if we want to add it we should add a matching `id` property on the actual label)
- I removed invalid `area-disabled` (if this was meant to be `aria-disabled` then it should be based on a property)
- I updated how `required` property is added, making it a proper boolean property.
- I changed `window.setTimeout` for `globalThis.setTimeout` so it's SSR safe
- I changed `messageNeeded.style.opacity = ` to string values, as expected by TS when removing the `any`

**Comment by eumarciel404** [comment]
> I went through your modals PR and made some changes / fixes but there's still work to do as well. Namely the `discord-message-needed-input` (now called `discord-text-input-message-needed-input`) doesn't show properly. The problem is that its div is nested in the `DiscordInputText` when it should be nested in the `DiscordModal` so that it can actually overlap the `dialog`'s content.
> 
> Also when clicking `cancel` or otherwise dismissing the form (i.e. through the submit handler) the warning should be removed again. This should be possible with the `disconnectedCallback` lifecycle method that we'd need to implement. See https://lit.dev/docs/components/lifecycle/#disconnectedcallback
> 
> Here is the problem shown in video:
> 
>  Screen.Recording.2024-08-18.at.13.49.03.mov 
> As for what I changed:
> 
> * Refactored out all the `:any`'s
> * You had some code comments in brazillian, I changed those to english
> * There was a typo in `discord-input-text-container` CSS where you had `flex: collmun;` instead of `flex-direction: column;`
> * Where you wrote `/* Here need danger color discord */` I got the proper colour
>   
>   * I also updated the colours to match Discord in a few other places
> * I have `discord-message-needed-input` a `width 50%` because adding the border I noticed it was too short for the text
>   
>   * I updated the text to match Discord
> * `_maxLengthCalc` and `_warn` were local values so I changed their decorator from `@property` to `@state` and made them `protected` because I want state to be accessible if someone decides to extend classes.
> * I removed the nested ternaries for the `textarea` values, using `when` instead.
> * I removed invalid `aria-labelledby` (if we want to add it we should add a matching `id` property on the actual label)
> * I removed invalid `area-disabled` (if this was meant to be `aria-disabled` then it should be based on a property)
> * I updated how `required` property is added, making it a proper boolean property.
> * I changed `window.setTimeout` for `globalThis.setTimeout` so it's SSR safe
> * I changed `messageNeeded.style.opacity = ` to string values, as expected by TS when removing the `any`

 Or i make equal when display are change to flex

**Review by favna — APPROVED** [review]

**Review by favna — APPROVED** [review]

**Pull request merged**